liblocation/src/main/java/com/mapbox/android/core/location/MapboxFusedLocationEngineImpl.java

```diff
@@ -0,0 +1,160 @@
+package com.mapbox.android.core.location;
+
+import android.app.PendingIntent;
+import android.content.Context;
+import android.content.Intent;
+import android.location.Location;
+import android.location.LocationListener;
+import android.location.LocationManager;
+import android.os.Bundle;
+import android.os.Looper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+import android.util.Log;
+
+import static com.mapbox.android.core.location.Utils.isBetterLocation;
+
+/**
+ * Mapbox replacement for Google Play Services Fused Location Client
+ * <p>
+ * Note: optimize engine logic for availability
+ */
+class MapboxFusedLocationEngineImpl extends AndroidLocationEngineImpl implements LocationListener {
+  private static final String TAG = "MapboxLocationEngine";
+
+  private LocationEngineCallback<LocationEngineResult> callback;
+  private Location currentBestLocation;
+
+  MapboxFusedLocationEngineImpl(@NonNull Context context) {
+    super(context);
+    this.callback = null;
+  }
+
+  @NonNull
+  @Override
+  LocationListener createListener(final LocationEngineCallback<LocationEngineResult> callback) {
```

Comment:
I'm a bit confused here... Why internal `createListener` is override and not `getLocationListener` to match overriden below `removeLocationListener`?